### PR TITLE
[misc] Add deactivate_all_snodes and ti.FieldsBuilder.deactivate_all

### PR DESCRIPTION
**Pull request by ljcc0930**
Related issue = #2936

### CODE REVIEW

**Comment by yuanming-hu** [comment]
Thanks - can we ...
1. Keep the old `ti.root.deactivate_all()` API?
2. Add a test for https://github.com/taichi-dev/taichi/issues/2936?

File: python/taichi/snode/fields_builder.py [hunk 1]
```diff
@@ -71,49 +71,60 @@ def empty(self):
...
+            warning(
+                """'deactivate_all()' would do nothing if FieldsBuilder is not finalized.
+    Due to 'ti.root' is implemented by 'ti.FieldsBuilder' implicitly, if you are
+    calling 'ti.root.deactivate_all()', please using 'ti.deactivate_all_snodes()'
+    instead.""")
+
     def dense(self, indices: Union[Sequence[_Axis], _Axis],
               dimensions: Union[Sequence[int], int]):
-        """Same as :func:`taichi.lang.SNode.dense`"""
+        """Same as :func:`taichi.lang.snode.SNode.dense`"""
```
Comment (k-ye) [inline python/taichi/snode/fields_builder.py:87]: have you tested the docstring link here?

Comment (ljcc0930) [reply python/taichi/snode/fields_builder.py:85]: It works

**Comment by ljcc0930** [comment]
> Thanks - can we ...
> 
> 1. Keep the old `ti.root.deactivate_all()` API?
> 2. Add a test for ["examples/features/sparse/taichi_bitmasked.py" no longer works #2936](https://github.com/taichi-dev/taichi/issues/2936)?

Fixed. Still, shall we deactivate all of the finalized snode trees after calling `ti.root.deactivate_all()`?

**Review by k-ye — APPROVED** [review]
LGTM!

**Pull request merged**